src/material/autocomplete/autocomplete-trigger.ts

```diff
@@ -858,6 +861,59 @@ export abstract class _MatAutocompleteTriggerBase
     // but the behvior isn't exactly the same and it ends up breaking some internal tests.
     overlayRef.outsidePointerEvents().subscribe();
   }
+
+  private _trackedModals = new Set<Element>();
+
+  /**
+   * Some browsers won't expose the accessibility node of the listbox overlay if there is an
+   * `aria-modal` and the overlay element is outside of it. This method works around the issue by
+   * pointing the `aria-owns` of all modals to the overlay element.
+   *
+   * While aria-owns is not required for the ARIA 1.2 `role="combobox"` interaction pattern,
+   * it fixes an issue with VoiceOver when the autocomplete appears inside of an `aria-model="true"`
+   * element (e.g. a dialog). Without this `aria-owns`, the `aria-modal` on a dialog prevents
+   * VoiceOver from "seeing" the autocomplete's listbox overlay for aria-activedescendant.
+   * Using `aria-owns` re-parents the autocomplete overlay so that it works again.
+   * See https://github.com/angular/components/issues/20694
+   */
+  private _exposeToModals() {
```

Comment:
```suggestion
  /**
   * If the autocomplete trigger is inside of an `aria-modal` element, connect
   * that modal to the options panel with `aria-owns`.
   *
   * For some browser +  screen reader combinations, when navigation is inside
   * of an `aria-modal` element, the screen reader treats everything outside
   * of that modal as hidden or invisible.
   * 
   * This causes a problem when the combobox trigger is _inside_ of a modal, because the
   * options panel is rendered _outside_ of that modal, preventing screen reader navigation
   * from reaching the panel.
   * 
   * We can work around this issue by applying `aria-owns` to the modal with the `id` of
   * the options panel. This effectively communicates to assistive technology that the
   * options panel is part of the same interaction as the modal.
   * 
   * At time of this writing, this issue is present in VoiceOver.
   * See https://github.com/angular/components/issues/20694
   */
  private _applyModalPanelOwnership() {
```

I've reworded the comment a bit, LMK if that's an accurate representation of the facts

One tip in general is that the first sentence of a function description should summarize what the function does, with additional context following. 

Also proposed renaming the function to be a bit more explicit, but feel free to iterate on other names

Reply:
Looks good to me, I added the comment and renamed the method